src/storage/accessors/FileDataAccessor.ts

```diff
@@ -188,7 +195,14 @@ export class FileDataAccessor implements DataAccessor {
     metadata.remove(RDF.terms.type, LDP.terms.Container);
     metadata.remove(RDF.terms.type, LDP.terms.BasicContainer);
     metadata.removeAll(DC.terms.modified);
-    metadata.removeAll(CONTENT_TYPE_TERM);
+    // When writing metadata for a document, do not remove the contentType when dealing with an unsupported media type.
+    // (unsupported media type: link.contentType is application/octect-stream and does not match metadata.contentType)
+    // This allows restoring the appropriate content type on data read (see getFileMetadata).
+    const isUnsupportedMediaTypeDocument = !isContainerPath(link.filePath) &&
+      link.contentType === APPLICATION_OCTET_STREAM && link.contentType !== metadata.contentType;
```

Comment:
I wonder if 
```suggestion
      link.contentType !== metadata.contentType;
```
is enough and whether that then would also cover cases like `application/activity+json` with a `.json` extension.

Reply:
Or even `application/ld+json; profile="https://www.w3.org/ns/activitystreams"`; that could be very exciting.

Reply:
Thanks! I'll look into it!

Reply:
`link.contentType !== metadata.contentType` should indeed be enough, but I don't think it's going to make any difference for the mentioned cases. The `mime` library simply does an exact string matching to map content-type to file extension and back. Things like `application/ld+json; profile="https://www.w3.org/ns/activitystreams"` should already work without this fix as the `contentType` value is `application/ld+json` and the extra parameters are already stored in metadata.